saito-core/src/core/consensus_thread.rs

```diff
@@ -138,6 +141,36 @@ impl ConsensusThread {
             );
         }
     }
+    async fn generate_issuance_tx(
+        &self,
+        mempool: Arc<RwLock<Mempool>>,
+        blockchain: Arc<RwLock<Blockchain>>,
+    ) {
+        info!("generating issuance init transaction");
+
+        let (blockchain, _blockchain_) = lock_for_read!(blockchain, LOCK_ORDER_BLOCKCHAIN);
```

Comment:
these locks should be moved below. just before 'for tx in txs' line. critical sections should be as small as possible